rpc/handlers.go

```diff
@@ -1120,3 +1123,48 @@ func (h *Handler) EstimateFee(broadcastedTxns []BroadcastedTransaction, id Block
 
 	return estimates, nil
 }
+
+func (h *Handler) TraceTransaction(hash felt.Felt) (TransactionTrace, *jsonrpc.Error) {
+	_, _, blockNumber, err := h.bcReader.Receipt(&hash)
+	if err != nil {
+		return nil, jsonrpc.Err(jsonrpc.InternalError, err.Error())
+	}
+	tx, err := h.bcReader.TransactionByHash(&hash)
+	if err != nil {
+		return nil, jsonrpc.Err(jsonrpc.InternalError, err.Error())
+	}
+
+	state, closer, err := h.bcReader.StateAtBlockNumber(blockNumber -1)
```

Comment:
@omerfirmak what if there is a request for tx that resides in 0 block? what should we do in this case

Reply:
Yes, that is an issue. Let's add a special case in `StateAtBlockNumber`. 

```
var EmptyState := core.NewState(db.NewMemTransaction)

if blockNumber == 0 {
   return EmptyState
}
```

Reply:
and in trace handler, we can just use `h.bcReader.StateAtBlockHash(block.ParentHash)` instead of `h.bcReader.StateAtBlockNumber(blockNumber -1)`

Reply:
Added